server/routers/user/user.py

```
@@ -44,136 +47,118 @@ def authenticate_user(user_name: str, user_password: str):
-def create_user(form_data: User):
-    with get_db() as (db, cursor):
-        cursor.execute("SELECT * FROM users WHERE user_name=%s", (form_data.user_name,))
-        user = cursor.fetchone()
-        if user:
-            return {"message": "User already exists."}
-
-        hashed_password = get_password_hash(form_data.user_password)
-        cursor.execute(
-            "INSERT INTO users (user_name, user_password, full_name, user_email) VALUES (%s, %s, %s, %s)",
-            (form_data.user_name, hashed_password, form_data.full_name, form_data.user_email),
+def create_user(form_data: User, session: Session = Depends(get_session)):
```

Ruff (0.15.13)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)